cli/command/trust/inspect_test.go

```diff
@@ -0,0 +1,380 @@
+package trust
+
+import (
+	"encoding/hex"
+	"io/ioutil"
+	"testing"
+
+	"github.com/docker/cli/cli/trust"
+	"github.com/docker/cli/internal/test"
+	"github.com/docker/cli/internal/test/testutil"
+	dockerClient "github.com/docker/docker/client"
+	"github.com/docker/notary"
+	"github.com/docker/notary/client"
+	"github.com/docker/notary/tuf/data"
+	"github.com/stretchr/testify/assert"
+)
+
+type fakeClient struct {
+	dockerClient.Client
+}
+
+func TestTrustInfoErrors(t *testing.T) {
+	testCases := []struct {
+		name          string
+		args          []string
+		expectedError string
+	}{
+		{
+			name:          "not-enough-args",
+			expectedError: "requires exactly 1 argument",
+		},
+		{
+			name:          "too-many-args",
+			args:          []string{"remote1", "remote2"},
+			expectedError: "requires exactly 1 argument",
+		},
+		{
+			name:          "sha-reference",
+			args:          []string{"870d292919d01a0af7e7f056271dc78792c05f55f49b9b9012b6d89725bd9abd"},
+			expectedError: "invalid repository name",
+		},
+		{
+			name:          "nonexistent-reg",
+			args:          []string{"nonexistent-reg-name.io/image"},
+			expectedError: "No signatures or cannot access nonexistent-reg-name.io/image",
+		},
+		{
+			name:          "invalid-img-reference",
+			args:          []string{"ALPINE"},
+			expectedError: "invalid reference format",
+		},
+		{
+			name:          "unsigned-img-reference",
+			args:          []string{"riyaz/unsigned-img"},
+			expectedError: "No signatures or cannot access riyaz/unsigned-img",
+		},
+		{
+			name:          "nonexistent-img-reference",
+			args:          []string{"riyaz/nonexistent-img"},
+			expectedError: "No signatures or cannot access riyaz/nonexistent-img",
+		},
+	}
+	for _, tc := range testCases {
+		cmd := newInspectCommand(
+			test.NewFakeCli(&fakeClient{}))
+		cmd.SetArgs(tc.args)
+		cmd.SetOutput(ioutil.Discard)
+		testutil.ErrorContains(t, cmd.Execute(), tc.expectedError)
+	}
+}
+
+func TestTrustInfo(t *testing.T) {
+	cli := test.NewFakeCli(&fakeClient{})
+	cmd := newInspectCommand(cli)
+	cmd.SetArgs([]string{"alpine"})
+	assert.NoError(t, cmd.Execute())
```

Comment:
Does this contact the "real" notary server?  Is that what you were referring to by fixture images in the description?

I think maybe `trust.GetNotaryRepository()` should be changed to `Cli.GetNotaryClient()` (or something like that) which would return an interface instead of a `client.NotaryRepostiory` struct . Once it's on the `command.Cli` interface it makes it really easy to fake it in tests.

Reply:
it does contact a real notary server. Is the idea that `Cli.GetNotaryClient` would wrap a notary repo with a subset of the API?  It seems that could certainly work

Reply:
It doesn't necessary have to wrap it with new structs, it could just return an Interface type instead of a struct, so that we can use a fake during testing.

Reply:
I've been trying to wrap my head around this but I'm not sure if it makes sense since a `NotaryRepository` is custom per-image.  Is that OK in the context of `command.Cli`, or are assuming that a `command.Cli` may persist for longer than a single image command?

Reply:
That does make it more difficult. We ran into this problem in #138 with the distribution client. In that case I wrapped the whole thing in an interface that was more appropriate for our use case in the cli (https://github.com/docker/cli/pull/138/files#diff-cb6e6f1ab385cb866626ee63bfa1a273R29).

It sounds like notary is using a similar pattern. 

I think something like this in `cli/command/cli.go` should work:

```go

func (cli *DockerCli) NotaryClient(ref trust.ImageRefAndAuth) trust.Repository {
    ...
}
```
where `trust.Repository` is the local interface we define (maybe in `cli/trust`).

We can still have multiple instances per `command.Cli`. In tests we can return a `fakeTrustRepository`.

Reply:
awesome, that would work.  I'll implement this 👍